### PR TITLE
DOXYGEN: Multiple improvements for the doxygen documentation

### PR DESCRIPTION
**Pull request by shamisp**
* Added support for bibtex
* New format for the front page
* Adding logos and figures
* Adding Introduction
* Adding Design

### CODE REVIEW

**Comment by shamisp** [comment]
@bosilca - please take a look

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/920/ for details.

**Comment by shamisp** [comment]
autobot build

**Comment by shamisp** [comment]
The failure is unrelated to the PR (system problem).

**Comment by bosilca** [comment]
:+1:

**Comment by shamisp** [comment]
@yosefe please take a look.

**Comment by shamisp** [comment]
@manjugv

File: doc/doxygen/intro.dox [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ * 
+ * \section Motivation
+ * The current state of the art in the world of communication middle ware
+ * typically takes two approaches.  First, communication packages such as
+ * Intel's (previously Qlogic) PSM, Mellanox's MXM, Portals, and IBM's PAMI
+ * provide flexible and high performance APIs for specific network vendors'
+ * hardware technologies.  As a result, multiple network technologies create
+ * significant challenges for programming model developers. While having
+ * multiple options is desirable, the ability to use the option that best fits
+ * a given communication task is highly desirable, regardless of which API it
```
Comment (yosefe) [inline doc/doxygen/intro.dox]: don't like the wording here - "desirable" appears twice

File: doc/doxygen/intro.dox [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ * We envision that these APIs will satisfy the networking needs of many
+ * programming models such as the Message Passing Interface (MPI), OpenSHMEM,
+ * Partitioned Global Address Space (PGAS) languages, task-based paradigms, and
+ * I/O bound applications.
+ *
+ * The design focuses on support for point-to-point communications
+ * (one-sided and two-sided), collective communication, and remote atomic
+ * operations. The initial UCX reference implementation is targeted to support
+ * current network technologies, such as: 
+ * + Open Fabrics - InfiniBand (Mellanox, Qlogic, IBM), libfabrics, iWARP, ROCEE
```
Comment (yosefe) [inline doc/doxygen/intro.dox]: RoCE

File: doc/doxygen/intro.dox [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ * hybrid applications that use both SHMEM and MPI programming models will be
+ * able to select between a single-shared UCX network context or a stand
+ * alone UCX network context for each one of them. Such flexibility,
+ * optimized resource sharing, and reduced memory consumption improve network
+ * and application performance.
+ * 
+ * Extreme-Scale scalability is an important goal of the project. To address
+ * it, UCX provides resolution for each one of the follow scalability
+ * requirements:
+ * + Minimal memory consumption - the design avoids order N (the number of processes) data structures usage and provides resource sharing capabilities for communication libraries.
```
Comment (yosefe) [inline doc/doxygen/intro.dox]: cam we wrap lines here?

Comment (shamisp) [inline doc/doxygen/intro.dox]: nope

**Pull request merged**